x-pack/plugins/ml/public/embeddables/anomaly_explorer/embeddable_explorer_container.tsx

```diff
@@ -0,0 +1,172 @@
+import { EuiCallOut, EuiLoadingChart, EuiResizeObserver, EuiText } from '@elastic/eui';
+import { Observable } from 'rxjs';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { throttle } from 'lodash';
+import { useExplorerInputResolver } from './explorer_input_resolver';
+import type { IAnomalyExplorerEmbeddable } from './anomaly_explorer_embeddable';
+import type {
+  AnomalyExplorerEmbeddableInput,
+  AnomalyExplorerEmbeddableOutput,
+  AnomalyExplorerEmbeddableServices,
+} from '..';
+import type { EntityField } from '../../../common/util/anomaly_utils';
+
+import { ExplorerAnomaliesContainer } from '../../application/explorer/explorer_charts_container';
+import { ML_APP_URL_GENERATOR } from '../../../common/constants/ml_url_generator';
+import { optionValueToThreshold } from '../../application/components/controls/select_severity/select_severity';
+import { ANOMALY_THRESHOLD } from '../../../common';
+import { UI_SETTINGS } from '../../../../../../src/plugins/data/common';
+import { TimeBuckets } from '../../application/util/time_buckets';
+import { EXPLORER_ENTITY_FIELD_SELECTION_TRIGGER } from '../../ui_actions/triggers';
+
+const RESIZE_THROTTLE_TIME_MS = 500;
+
+export interface EmbeddableExplorerContainerProps {
+  id: string;
+  embeddableContext: InstanceType<IAnomalyExplorerEmbeddable>;
+  embeddableInput: Observable<AnomalyExplorerEmbeddableInput>;
+  services: AnomalyExplorerEmbeddableServices;
+  refresh: Observable<any>;
+  onInputChange: (input: Partial<AnomalyExplorerEmbeddableInput>) => void;
+  onOutputChange: (output: Partial<AnomalyExplorerEmbeddableOutput>) => void;
+}
+
+export const EmbeddableExplorerContainer: FC<EmbeddableExplorerContainerProps> = ({
+  id,
+  embeddableContext,
+  embeddableInput,
+  services,
+  refresh,
+  onInputChange,
+  onOutputChange,
+}) => {
+  const [chartWidth, setChartWidth] = useState<number>(0);
+  const [severity, setSeverity] = useState(optionValueToThreshold(ANOMALY_THRESHOLD.MINOR));
+
+  const [
+    { uiSettings },
+    {
+      data: dataServices,
+      share: {
+        urlGenerators: { getUrlGenerator },
+      },
+      uiActions,
+    },
+  ] = services;
+  const { timefilter } = dataServices.query.timefilter;
+
+  const mlUrlGenerator = useMemo(() => getUrlGenerator(ML_APP_URL_GENERATOR), [getUrlGenerator]);
+
+  const timeBuckets = useMemo(() => {
+    return new TimeBuckets({
+      'histogram:maxBars': uiSettings.get(UI_SETTINGS.HISTOGRAM_MAX_BARS),
+      'histogram:barTarget': uiSettings.get(UI_SETTINGS.HISTOGRAM_BAR_TARGET),
+      dateFormat: uiSettings.get('dateFormat'),
+      'dateFormat:scaled': uiSettings.get('dateFormat:scaled'),
+    });
+  }, []);
+
+  const { chartsData, isLoading: isExplorerLoading, error } = useExplorerInputResolver(
+    embeddableInput,
+    onInputChange,
+    refresh,
+    services,
+    chartWidth,
+    severity.val
+  );
+  const resizeHandler = useCallback(
```

Comment:
You can try [useThrottleFn](https://github.com/streamich/react-use/blob/master/docs/useThrottle.md) instead

Reply:
I'm not seeing this being used anywhere in Kibana currently. Not sure if the cost of importing it outweighs the shortened form so I'm going keep it as is.

Reply:
We use `react-use` lib in quite some places, it's existing Kibana dependency.